### PR TITLE
[minor] #MAXMIS-748: add role to auto configure image stitching

### PR DESCRIPTION
**Pull request by bmoferrall**
Jira items:

https://jsw.ibm.com/browse/MAXMIS-747
and
https://jsw.ibm.com/browse/MAXMIS-748

[Complementary pytest suite, with a successful end-to-end run: [link](https://github.ibm.com/maximoappsuite/fvt-ibm-mas-manage/pull/488).]

Civil wants to remove manual steps as described in the [documentation](https://www.ibm.com/docs/en/mci/continuous-delivery?topic=application-configuring-image-stitching-service) (the steps are wrong anyway).

Add role to auto-deploy the image stitching application when Civil is installed
- Check if Civil is installed by inspecting the spec.components section of the ManageWorkspace CR. Skip all steps if not
- Check if PVC has been created. Create if not
- Patch ManageWorkspace CR with PV specifications
- Patch ManageWorkspace CR with PVC name
- Configure two system properties required for the image stitching application

To test the role was deployed to a test pod (using latest cli image) in Civil's FVT cluster and run from there.

I've run several tests on a pfvt cluster, described in later comments below

### CODE REVIEW

File: ibm/mas_devops/roles/suite_manage_imagestitching_config/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- hosts: localhost
+  any_errors_fatal: true
+  vars:
+    mas_instance_id: civil
+    mas_workspace_id: masdev
+    mas_domain: civil.ibmmasdev.com
+    stitching_pvcname: manage-imagestitching
+    stitching_storage_class: nfs-client
+    stitching_storage_size: 20Gi
+    stitching_storage_mode: ReadWriteMany
```
Comment (MaeveOReilly) [inline ibm/mas_devops/roles/suite_manage_imagestitching_config/README.md:73]: There is a testCvScan[NoReadWriteMany] [rule](https://github.ibm.com/IBMPrivateCloud/content-verification/blob/main/docs/rules/NoReadWriteMany.md) + [test](https://dashboard.masdev.wiotp.sl.hursley.ibm.com/tests/fvt90x/testsuite/ibm-mas-manage/contentverification) which we were failing.  Leonardo fixed that one  - https://github.ibm.com/maximoappsuite/ibm-mas-imagestitching/pull/68/files
Is there a reason to have ReadWriteMany?  I think Once should be better for our use case.

**Review by MaeveOReilly — COMMENTED** [review]
Apart from my concern with using ReadWriteMany looks great to me.  Would be good to give it a test in a pfvt as well.  Fyre and Roks are not always consistent esp w.r.t. PVC stuff.

File: ibm/mas_devops/roles/suite_manage_imagestitching_config/README.md (hunk 1 above)
Comment (bmoferrall) [reply ibm/mas_devops/roles/suite_manage_imagestitching_config/README.md:73]: My understanding is that the Manage pod and the Imagestitching pod will both write to the PV. Have we tested with ReadWriteOnce? Maybe I'm misunderstanding the difference

**Review by MaeveOReilly — APPROVED** [review]

File: ibm/mas_devops/roles/suite_manage_imagestitching_config/README.md (hunk 1 above)
Comment (bmoferrall) [reply ibm/mas_devops/roles/suite_manage_imagestitching_config/README.md]: Updated. I'll test later with pvt if I can figure out the issue with attachments

**Comment by bmoferrall** [comment]
While this works fine on fyre environments it's not fully working on an fvt-personal environment.
The ManageWorkspace and ManageDeployment CRs are updating correctly:

```
    deployment:
      persistentVolumes:
        - accessModes:
            - ReadWriteMany
          mountPath: /DOCLINKS
          pvcName: manage-doclinks
          size: 20Gi
          storageClassName: ibmc-file-gold-gid
        - accessModes:
            - ReadWriteMany
          mountPath: /bim
          pvcName: manage-bim
          size: 20Gi
          storageClassName: ibmc-file-gold-gid
        - accessModes:
            - ReadWriteMany
          mountPath: /jms
          pvcName: manage-jms
          size: 20Gi
          storageClassName: ibmc-file-gold-gid
        - accessModes:
            - ReadWriteOnce
          mountPath: /imagestitching
          pvcName: manage-imagestitching
          size: 20Gi
          storageClassName: **ibmc-block-gold**
```
However, the Deployment config, and hence the Manage pod, is not updating.
Since the other PVs are using ReadWriteMany I'm going to revert to that and retry

**Comment by bmoferrall** [comment]
Adding the wait task and switching back to ReadWriteMany seems to have resolved the issue on the IBM Cloud personal fvt env:
[app-cfg-manage.log](https://github.com/user-attachments/files/17618155/app-cfg-manage.log)

```
TASK [ibm.mas_devops.suite_manage_imagestitching_config : Get stitching status] ***
ok: [localhost] => changed=false 
  ansible_facts:
    stitchingDeployed: true
```
manage UI pod:
```
      volumeMounts:
        - name: manage-truststore
          readOnly: true
          mountPath: /truststore
        - name: pfvtstitch-masdev-cert-public-81
          readOnly: true
          mountPath: /etc/ssl/certs/public-manage-tls
        - name: pfvtstitch-internal-manage-tls
          readOnly: true
          mountPath: /etc/ssl/certs/internal-manage-tls
        - name: manage-doclinks
          mountPath: /DOCLINKS
        - name: manage-bim
          mountPath: /bim
        - name: manage-jms
          mountPath: /jms
*      - name: manage-imagestitching
*         mountPath: /imagestitching
        - name: bundle-properties
          mountPath: /config/manage/properties
        - name: ibm-entitlement
          mountPath: /config/manage/ibm-entitlement
        - name: jvm-options
          mountPath: /config/manage/jvmoptions
        - name: additionalserverconfig
          mountPath: /config/manage/serverxml
        - name: dbdriver-config
          mountPath: /config/manage/customdbdriver
        - name: kube-api-access-tkx75
          readOnly: true
          mountPath: /var/run/secrets/kubernetes.io/serviceaccount

```
There was a small typo in the pytest suite, so I'll do another run later to fully validate

**Review by MaeveOReilly — DISMISSED** [review]

**Comment by bmoferrall** [comment]
I had a successful run yesterday on pfvt:
https://dashboard.masdev.wiotp.sl.hursley.ibm.com/tests/pfvtstitch/testsuite/ibm-mas-manage/civil-api-defectdetection

This morning I configured attachments as an additional validation. Unfortunately the stitch redeployment took 35 mins. Not sure why it takes so long for changes to filter down on pfvt environments. I've bumped up the wait time to 60 mins and have kicked off another run with attachments configured: https://github.ibm.com/maximoappsuite/fvt-personal/blob/stitch/.travis.yml#L111

**Comment by bmoferrall** [comment]
Another successful run. The pipeline is still in progress as I write, but the attachments were set up correctly and the end-to-end stitching test was successful:

https://dashboard.masdev.wiotp.sl.hursley.ibm.com/tests/pfvtstitch/testsuite/ibm-mas-manage/civil-api-defectdetection

afbeelding

**Review by alequint — APPROVED** [review]
Looks good to me as both actions will patch ManageWorkspace in the same role/task and same checking will be done to verify if CR has been redeployed. Let's keep an eye in our pipelines, how this runs behave daily.

**Pull request merged**